File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/MessageCell.m
//
//  MsgCell.m
//  OpenVideoCall
//
//  Created by 3 on 2020/12/15.
//  Portions Copyright (c) 2020 meta-rti. All rights reserved.
//

#import "MessageCell.h"
#import "CommonExtension.h"

@interface MessageCell ()
@property (weak, nonatomic) IBOutlet UIView *colorView;
@property (weak, nonatomic) IBOutlet UILabel *messageLabel;
@end

@implementation MessageCell
- (void)setMessage:(Message *)message {
    self.colorView.backgroundColor = (message.type == MessageTypeInfo ? UIColor.WJMessageInfoColor : UIColor.WJMessageErrorColor);
    self.messageLabel.text = message.text;
}
@end

File: Group-Video/OpenVideoCall-iOS/OpenVideoCall/MessageCell.swift
//
//  ChatMessageCell.swift
//  OpenVideoCall
//
//  Created by 3 on 2020/12/16.
//  Portions Copyright (c) 2020 meta-rti. All rights reserved.
//

import UIKit

class MessageCell: UITableViewCell {
    
    @IBOutlet weak var colorView: UIView!
    @IBOutlet weak var messageLabel: UILabel!
    
    func set(with message: Message) {
        colorView.backgroundColor = message.type.color()
        messageLabel.text = message.text
    }
}
